src/core/credentials/transport/tls/grpc_tls_certificate_provider.h

```diff
@@ -208,6 +210,62 @@ class FileWatcherCertificateProvider final
   std::map<std::string, WatcherInfo> watcher_info_ ABSL_GUARDED_BY(mu_);
 };
 
+// Implements a provider that uses in-memory data that can be modified in a
+// thread-safe manner.
+class InMemoryCertificateProvider final : public grpc_tls_certificate_provider {
+ public:
+  static std::shared_ptr<InMemoryCertificateProvider> CreateCertificateProvider(
+      std::string root_certificates, PemKeyCertPairList pem_key_cert_pairs);
+  InMemoryCertificateProvider(std::string root_certificates,
```

Comment:
I don't think we should pass in the cert data via the ctor.  Instead, callers should just instantiate the provider and then call the `UpdateRoot()` and `UpdateIdentity()` methods to set the data.  That way, there's only one code-path to do this.